### DISCUSSION TITLE
TESTed - Python: how to test multiple outputs of a function?

### DISCUSSION THREAD

**Discussion by marcothijs** [dodona-edu/feedback-and-support#32] [Questions - Vragen] [answered]
![Image 28-02-2024 at 10 23](https://github.com/dodona-edu/dodona/assets/60760890/12a7e043-7936-4bee-96ec-6b25d3611cc4)
Hi, 

Please check out https://dodona.be/nl/activities/1805655600/
I want to test the function that has 4 outputs (returns 4 strings). I see that it shows as a tuple with (  ,  ,  , ) but it expect to have " "
around it where it doesn't show them in my output, so the test fails.
lijn 10 and 11 in the yaml script:
- expression: "ipv4_informatie('192.168.1.10')"
   return: ('C', '192.168.1.0', '255.255.255.0', '192.168.1.255')

I've put the lines in comment so the rest of the script will test ok. 

How to fix this? 
Thanks in advance.
Marco

**Comment by niknetniko** [comment, marked as answer]
YAML only supports lists, so you need to use tags to "cast" the value to the correct type:

```yaml
- expression: "ipv4_informatie('192.168.1.10')"
  return: !tuple ['C', '192.168.1.0', '255.255.255.0', '192.168.1.255']
```

**Reply by marcothijs** [reply]
Thanks! That does the job. TESTed and YAML are not my friends ;-). I have an other excercise I run into test issue, but I'll make a new question for that.
